conan/tools/meson/meson.py

```diff
@@ -1,272 +1,52 @@
-                          'ppc64': ('ppc64', 'ppc64', 'big'),
-                          's390': ('s390', 's390', 'big'),
-                          's390x': ('s390x', 's390x', 'big'),
-                          'sh4le': ('sh4', 'sh4', 'little'),
-                          'sparc': ('sparc', 'sparc', 'big'),
-                          'sparcv9': ('sparc64', 'sparc64', 'big'),
-                          'wasm': ('wasm32', 'wasm32', 'little'),
-                          'x86': ('x86', 'x86', 'little'),
-                          'x86_64': ('x86_64', 'x86_64', 'little')}
-        system = system_map.get(machine_os, machine_os.lower())
-        default_cpu_tuple = (machine_arch.lower(), machine_arch.lower(), 'little')
-        cpu_tuple = cpu_family_map.get(machine_arch, default_cpu_tuple)
-        cpu_family, cpu, endian = cpu_tuple[0], cpu_tuple[1], cpu_tuple[2]
-        context = {
-            'system': self._to_meson_value(system),
-            'cpu_family': self._to_meson_value(cpu_family),
-            'cpu': self._to_meson_value(cpu),
-            'endian': self._to_meson_value(endian),
-        }
-        return self._render(self._machine_template, context)
-
-    @property
-    def _cross_content(self):
-        os_build, arch_build, os_host, arch_host = get_cross_building_settings(self._conanfile)
-        os_target, arch_target = os_host, arch_host  # TODO: assume target the same as a host for now?
-
-        build_machine = self._to_meson_machine(os_build, arch_build)
-        host_machine = self._to_meson_machine(os_host, arch_host)
-        target_machine = self._to_meson_machine(os_target, arch_target)
-
-        context = self._context
-        context['build_machine'] = build_machine
-        context['host_machine'] = host_machine
-        context['target_machine'] = target_machine
-        return self._render(self._cross_file_template, context)
-
-    def _write_native_file(self):
-        save(self._native_filename, self._native_content)
-
-    def _write_cross_file(self):
-        save(self._cross_filename, self._cross_content)
-
-    def generate(self):
+    def _build_dir(self):
+        build = self._conanfile.build_folder
+        if self._build_folder:
+            build = os.path.join(self._conanfile.build_folder, self._build_folder)
+        return build
+
+    def configure(self, source_folder=None):
+        source = self._conanfile.source_folder
+        if source_folder:
+            source = os.path.join(self._conanfile.source_folder, source_folder)
+
+        cmd = "meson setup"
         if cross_building(self._conanfile):
-            self._write_cross_file()
+            cmd += ' --cross-file "{}"'.format(MesonToolchain.cross_filename)
         else:
-            self._write_native_file()
+            cmd += ' --native-file "{}"'. format(MesonToolchain.native_filename)
+        cmd += ' "{}" "{}"'.format(self._build_dir, source)
+        if self._conanfile.package_folder:
```

Comment:
in some situations (which?) `package folder` is `None`, which results in `-Dprefix=None` and meson fails

Reply:
Local flow or ``test_package`` might not have package folder?

Reply:
yes, it was a `test_package`. is it expected?

Reply:
Yes. It has been merged for 1.33 in https://github.com/conan-io/conan/pull/8117